clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -581,6 +593,13 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
     public List<ClientResponse> poll(long timeout, long now) {
         ensureActive();
 
+        if (!this.isBootstrapped() && !this.isBootstrapDisabled()) {
```

Comment:
let's move this to ensureBootstrapped().

I also wonder if we could make bootstrapState and optional, so we just need to check bootstrapState.isEmpty() - true: don't need to bootstrap, false: perform bootstrap